src/groupBy.test.ts

```diff
@@ -176,6 +176,56 @@ describe("typing", () => {
     >();
   });
 
+  test("symbol", () => {
+    const mySymbol = Symbol("mySymbol");
+    const data = groupBy(
+      [
+        { [mySymbol]: "cat", b: 123 },
+        { [mySymbol]: "dog", b: 456 },
+      ],
+      () => mySymbol,
+    );
+    expectTypeOf(data).toEqualTypeOf<
+      Partial<
+        Record<
+          typeof mySymbol,
+          NonEmptyArray<{ [mySymbol]: string; b: number }>
+        >
+      >
+    >();
+  });
+
+  test("branded string", () => {
+    type Branded<K, T> = K & { _type: T };
+    type UserID = Branded<string, "UserId">;
+
+    const data = groupBy(
+      [
+        { owner: "U1" as UserID, name: "Phone" },
+        { owner: "U2" as UserID, name: "Pencil" },
+      ],
+      (x) => x.owner,
```

Comment:
use prop?
```suggestion
      prop("owner"),
```

Reply:
test removed, so this is outdated—i didn't use prop initially for consistency